src/App.jsx

```diff
@@ -14,70 +16,68 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const clearGood = () => {
+    setSelectedGood('');
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  return (
+    <main className="section container">
+      {selectedGood
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
             <button
-              data-cy="AddButton"
+              onClick={clearGood}
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              className="delete ml-3"
+            />
+          </h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )
+      }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelected = selectedGood === good;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={isSelected && 'has-background-success-light'}
+              >
+                <td>
+                  <button
+                    onClick={
+                      () => setSelectedGood(isSelected ? clearGood : good)
```

Comment:
Currently, when `isSelected` == true, you are passing `clearGood` to `setSelectedGood`. So we are saving function into state. This doesn't break the app, but it is wrong, and may cause trouble in the future
```suggestion
                      () => setSelectedGood(isSelected ? "" : good)
```